220080/문제5.py

```diff
@@ -0,0 +1,21 @@
+# 암호생성기
+from collections import deque
+import sys
+sys.stdin = open('input.txt','r')
+
+
+num = int(input())
+num1 = list(map(int,input().split()))
+q = deque(num1)
+
+print(num1[0],num1[-1])
+while 1:
+    for i in range(1,6):
+        q[0] = q[0]-i
+        q.append(q.popleft())
+        if q[-1] <= 0:
+            break
+    if q[-1] == [0]:
+        break
+
+print(q)
```

Comment:
덱 활용 아주 좋습니다!  👍 
이제 문제 조건의 '0보다 작은 수는 0으로 처리하기' 를 구현해주시고, 문제 상황에 맞게 입력 값을 받아주시면 해결하실 수 있을 거에요.
```suggestion
    if q[-1] == 0:
        break

print(q)
```
마지막 원소가 0인지 체크하려면, 위와 같이 작성해주는 것이 맞습니다!